src/cript/api/api.py

```diff
@@ -675,6 +677,11 @@ def download_file(self, object_name: str, destination_path: str = ".") -> None:
             just downloads the file to the specified path
         """
 
+        # if the file source is a URL
+        if object_name.startswith("http"):
```

Comment:
I am a little confused, don't AWS S3 files also start with `http`?

Reply:
nope no longer.

the way I used to retrieve the objects from the AWS S3 cloud storage was to convert what I just uploaded into a link and save that into the file source. However, I later found that the file link expires after a day or so. 

The way that both frontend and SDK grabs the files from AWS S3 is by getting them via their object_names such as `/python_sdk_files/my_uploaded_file.txt`

so now the file node source can be:
**file node local file source:** `/python_sdk_files/my_uploaded_file.txt`
**file node url source:** `https://my-file-link.pdf`

Reply:
see `file.py` `is_local`

Reply:
oooh you are right!

should we merge this in because it is about downloading web file source and then tackle the refactoring in a separate PR?

Reply:
Up to you, I am not a big fan of merging broken code, but sometimes that is necessary.